magenta-lib/src/main/scala/magenta/tasks/ASGTasks.scala

```diff
@@ -244,6 +248,31 @@ case class CullInstancesWithTerminationTag(
 
             isTaggedForTermination && !isAlreadyTerminating
           })
+
+        val instancesToRetain = allInstances.diff(instancesToKill).toList
+        resources.reporter.verbose(
+          s"Decided to keep the following instances: ${instancesToRetain.map(_.instanceId).mkString(", ")}"
+        )
+
+        if (instancesToRetain.size != instancesToKill.size) {
+          resources.reporter.warning(
+            s"Terminating ${instancesToKill.size} instances and retaining ${instancesToRetain.size} instances"
+          )
+          logger.warn(
```

Comment:
We should get this log line in ELK which will make it easier to ascertain frequency of the problem (and whether services other than `big-loader` are affected)